packages/SwingSet/src/kernel/vatManager/vat-warehouse.js

```diff
@@ -173,29 +245,66 @@ export function makeVatWarehouse(kernelKeeper, vatLoader, policyOptions) {
    * @param {string} currentVatID
    */
   async function applyAvailabilityPolicy(currentVatID) {
-    // console.log('applyAvailabilityPolicy', currentVatID, recent);
-    const pos = recent.indexOf(currentVatID);
-    // console.debug('applyAvailabilityPolicy', { currentVatID, recent, pos });
-    // already most recently used
-    if (pos + 1 === maxVatsOnline) return;
-    if (pos >= 0) recent.splice(pos, 1);
-    recent.push(currentVatID);
-    // not yet full
-    if (recent.length <= maxVatsOnline) return;
-    const [lru] = recent.splice(0, 1);
-    // console.debug('evicting', { lru });
+    const lru = recent.add(currentVatID);
+    if (!lru) {
+      return;
+    }
+    // const {
+    //   options: { description, managerType },
+    // } = ephemeral.vats.get(lru) || assert.fail();
+    // console.info('evict', lru, description, managerType, 'for', currentVatID);
     await evict(lru);
   }
 
+  /** @type { string | undefined } */
+  let lastVatID;
+
   /** @type {(vatID: string, d: VatDeliveryObject) => Promise<Tagged> } */
   async function deliverToVat(vatID, delivery) {
     await applyAvailabilityPolicy(vatID);
-    const recreate = true; // PANIC in the failure case
+    lastVatID = vatID;
 
+    const recreate = true; // PANIC in the failure case
     const { manager } = await ensureVatOnline(vatID, recreate);
     return manager.deliver(delivery);
   }
 
+  /**
+   * Save a snapshot of most recently used vat,
+   * depending on snapshotInterval.
+   */
+  async function maybeSaveSnapshot() {
+    if (!lastVatID || !lookup(lastVatID)) {
+      return false;
+    }
+
+    const recreate = true; // PANIC in the failure case
+    const { manager } = await ensureVatOnline(lastVatID, recreate);
+    if (!manager.makeSnapshot) {
+      return false;
+    }
+
+    const vatKeeper = kernelKeeper.provideVatKeeper(lastVatID);
+    let reason;
+    const current = vatKeeper.getTranscriptEndPosition().itemCount;
+    if (snapshotInitial === current) {
+      reason = { snapshotInitial };
+    } else {
+      const lastSnapshot = vatKeeper.getLastSnapshot();
+      const done = lastSnapshot ? lastSnapshot.startPos.itemCount : 0;
+      if (current - done >= snapshotInterval) {
```

Comment:
this is calling out for a vatKeeper method, `getTranscriptLength` or `getReplayableTranscriptLength` or something

Reply:
"truncated" transcript length?